unit_tests/testVTUFiles/testVTUFiles.f90

```diff
@@ -0,0 +1,232 @@
+      CALL SetupTest1_Mesh
+      CALL testVTUFile%writeMesh(testVTKMesh) ! Try to write STRUCTURED_POINTS
+      bool = .NOT.testVTUFile%hasMesh
+      ASSERT(bool, 'testVTUFile%writeMesh(...) STRUCTURED_POINTS')
+      CALL testVTUFile%clear(.TRUE.)
+    ENDSUBROUTINE testFailSTRUCTURED_POINTS
+!
+!-------------------------------------------------------------------------------
+    SUBROUTINE testUNSTRUCTURED_GRID
+      !Test writeMesh for UNSTRUCTURED_GRID
+      CALL testVTUFile%initialize(666,'testVTU2.vtu')
+      CALL SetupTest2_Mesh
+      CALL testVTUFile%writeMesh(testVTKMesh)
+      bool = testVTUFile%hasMesh
+      ASSERT(bool, 'testVTUFile%writeMesh(...) UNSTRUCTURED_GRID')
+
+      !Write data on the mesh
+      CALL SetupTest2_Data
+      CALL testVTUFile%writeScalarData(testVTKData)
+      testVTKData%varname='mat_val'
+      testVTKData%vtkDataFormat='float'
+      CALL testVTUFile%writeScalarData(testVTKData)
+
+      !Test writeScalarData
+      testVTKData%vtkDataFormat='long'
+      testVTKData%varname='mat_long'
+      CALL testVTUFile%writeScalarData(testVTKData) !repeat for long
+      testVTKData%vtkDataFormat='short'
+      testVTKData%varname='mat_short'
+      CALL testVTUFile%writeScalarData(testVTKData) !repeat for short
+      testVTKData%vtkDataFormat='int'
+      testVTKData%varname='mat_float'
+      CALL testVTUFile%writeScalarData(testVTKData) !write as int
+      testVTKData%vtkDataFormat='double'
+      testVTKData%varname='mat_double'
+      CALL testVTUFile%writeScalarData(testVTKData) !write as double
+      CALL testVTUFile%close
+      CALL testVTUFile%clear
+    ENDSUBROUTINE testUNSTRUCTURED_GRID
+!
+!-------------------------------------------------------------------------------
+    SUBROUTINE testMultiVTU
+      !Test writeMesh for UNSTRUCTURED_GRID
+      CALL testPVTUFile%clear
+      CALL testPVTUFile%initialize(666,'testPVTU_0.vtu')
+      CALL SetupTest2_Mesh
+      CALL testPVTUFile%writeMesh(testVTKMesh)
+      bool = testPVTUFile%hasMesh
+      ASSERT(bool,'testPVTUFile%writeMesh(...) UNSTRUCTURED_GRID 0')
+      !Write data on the mesh
+      CALL SetupTest2_Data
+      CALL testPVTUFile%writeScalarData(testVTKData)
+      CALL testPVTUFile%close
+      CALL testPVTUFile%clear
+
+      !Write to second file
+      CALL testPVTUFile%initialize(666,'testPVTU_1.vtu')
+      !Translate mesh
+      CALL testVTKMesh%translate(-4.0_SRK,-5.0_SRK,3.0_SRK)
+      CALL testPVTUFile%writeMesh(testVTKMesh)
+      bool = testPVTUFile%hasMesh
+      ASSERT(bool,'testPVTUFile%writeMesh(...) UNSTRUCTURED_GRID 1')
+      !Write data on the mesh
+      CALL testPVTUFile%writeScalarData(testVTKData)
+      CALL testPVTUFile%close
+      CALL testPVTUFile%clear
+
+      CALL testPVTUFile%writepvtu(666,'testPVTU',2)
+    ENDSUBROUTINE testMultiVTU
+!
+!-------------------------------------------------------------------------------
+    SUBROUTINE SetupTest1_Mesh
```

Comment:
parentheses for function/subroutine definitions also, even when there are no arguments.

reasoning is to be able to differentiate variables from functions